File: web-frontend/src/services/aiPromptService.ts
/**
 * AI Prompt Management Service
 *
 * Provides CRUD access to the three organizer-editable OpenAI prompts:
 *   - event_description  — GPT-4o event description generation
 *   - theme_image        — DALL-E theme image generation
 *   - abstract_quality   — GPT-4o abstract quality review
 *
 * CRITICAL: Always use this service — never call apiClient directly from components.
 */

import apiClient from '@/services/api/apiClient';
import type { components } from '@/types/generated/events-api.types';

export type AiPromptResponse = components['schemas']['AiPromptResponse'];
export type UpdateAiPromptRequest = components['schemas']['UpdateAiPromptRequest'];

export const listAiPrompts = async (): Promise<AiPromptResponse[]> => {
  const response = await apiClient.get<AiPromptResponse[]>('/ai-prompts');
  return response.data;
};

export const getAiPrompt = async (promptKey: string): Promise<AiPromptResponse> => {
  const response = await apiClient.get<AiPromptResponse>(`/ai-prompts/${promptKey}`);
  return response.data;
};

export const updateAiPrompt = async (
  promptKey: string,
  promptText: string
): Promise<AiPromptResponse> => {
  const body: UpdateAiPromptRequest = { promptText };
  const response = await apiClient.put<AiPromptResponse>(`/ai-prompts/${promptKey}`, body);
  return response.data;
};

export const resetAiPrompt = async (promptKey: string): Promise<AiPromptResponse> => {
  const response = await apiClient.post<AiPromptResponse>(`/ai-prompts/${promptKey}/reset`);
  return response.data;
};

File: services/event-management-service/src/main/java/ch/batbern/events/service/AiPromptService.java
package ch.batbern.events.service;

import ch.batbern.events.domain.AiPrompt;
import ch.batbern.events.repository.AiPromptRepository;
import jakarta.persistence.EntityNotFoundException;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * Service for managing organizer-editable AI prompts (Story 10.x).
 *
 * Three prompts are seeded by migration V82:
 *   - event_description  — GPT-4o event description generation
 *   - theme_image        — DALL-E theme image generation
 *   - abstract_quality   — GPT-4o abstract quality review
 *
 * Used by BatbernAiService to retrieve the current prompt text at call time.
 */
@Service
@RequiredArgsConstructor
@Slf4j
public class AiPromptService {

    private final AiPromptRepository repository;

    public List<AiPrompt> findAll() {
        return repository.findAll();
    }

    public AiPrompt findByKey(String promptKey) {
        return repository.findById(promptKey)
                .orElseThrow(() -> new EntityNotFoundException("AI prompt not found: " + promptKey));
    }

    /**
     * Returns the current prompt text for the given key.
     * Used internally by BatbernAiService. Falls back to empty string if key missing.
     */
    public String getPromptText(String promptKey) {
        return repository.findById(promptKey)
                .map(AiPrompt::getPromptText)
                .orElse("");
    }

    @Transactional
    public AiPrompt update(String promptKey, String promptText) {
        AiPrompt prompt = findByKey(promptKey);
        prompt.setPromptText(promptText);
        log.info("AI prompt '{}' updated", promptKey);
        return repository.save(prompt);
    }

    @Transactional
    public AiPrompt reset(String promptKey) {
        AiPrompt prompt = findByKey(promptKey);
        prompt.setPromptText(prompt.getDefaultText());
        log.info("AI prompt '{}' reset to default", promptKey);
        return repository.save(prompt);
    }
}
